### PR TITLE
Port GDA logic from pcli

### LINKED ISSUE #981: Client-side mechanism for continuous GDA
GDA pricing mechanism:

10m: 4 auctions, 60 blocks long (5m), step_count=60, start heights ~ poisson(lambda = 0.775/min), rel gas = 480x
30m: 12 auctions, 60 blocks long (5m), step_count=60, start heights ~ poisson(lambda = 0.607/min), rel gas = 1440x
1h: 12 auctions, 120 blocks long (10m), step_count=60, start heights ~ poisson(lambda = 0.303/min), rel gas = 1440x
2h: 24 auctions, 120 blocks long (10m), step_count=60, start heights ~ poisson(lambda = 0.272/min), rel gas = 2880x
6h: 36 auctions, 240 blocks long (20m), step_count=60, start heights ~ poisson(lambda = 0.129/min), rel gas = 4320x
12h: 36 auctions, 480 blocks long (40m), step_count=60, start heights ~ poisson(lambda = 0.064/min), rel gas = 4320x
24h: 48 auctions, 720 blocks long (1h), step_count=60, start heights ~ poisson(lambda = 0.042/min), rel gas = 5760x
48h: 48 auctions, 1440 blocks long (2h), step_count=60, start heights ~ poisson(lambda = 0.021/min), rel gas = 5760x

Source: [Discord](https://discord.com/channels/824484045370818580/938328380917559326/1235723630965686363)
pcli reference: https://github.com/penumbra-zone/penumbra/pull/4362

### PR DESCRIPTION
**Pull request by jessepinho**
This PR brings over the logic from pcli for generating sub-auctions for a given auction configuration.

https://github.com/penumbra-zone/web/assets/1121544/ef5f91de-6186-4f74-9f2e-5f7800230588

(We're going to need some help with the transaction approval UI, I think...)

🚨 Reviewers, please review the poission distribution function, as there's a good chance it's incorrect.

## In this PR
- Create a `getSubAuctions()` helper that generates sub-auctions to add to the `TransactionPlannerRequest`.
- Tweak the duration slider on the auction form; now, it shows how many sub-auctions will be created.
- Adjust the step count to 60 across all auction durations, to match pcli.

## Future PR
- Filter out auction NFTs from the auction/swap asset selectors

Closes #981

### CODE REVIEW

File: apps/minifront/src/state/dutch-auction/assemble-schedule-request.test.ts
Comment (jessepinho) [inline apps/minifront/src/state/dutch-auction/assemble-schedule-request.test.ts:55]: Most of the logic in `assembleScheduleRequest()` has been extracted to `getSubAuctions`; thus, so too have the tests.

File: apps/minifront/src/state/dutch-auction/assemble-schedule-request.ts
Comment (jessepinho) [inline apps/minifront/src/state/dutch-auction/assemble-schedule-request.ts:31]: all this logic now lives in `getSubAuctions()`

File: apps/minifront/src/components/swap/dutch-auction/duration-slider.tsx
Comment (jessepinho) [inline apps/minifront/src/components/swap/dutch-auction/duration-slider.tsx:35]: screencap

File: packages/types/src/amount.ts
Comment (jessepinho) [inline packages/types/src/amount.ts:30]: Without this rounding, JS throws because we're trying to convert a decimal to a `BigInt`.

**Review by grod220 — APPROVED** [review]
I don't know much about the Poisson stuff, before merging let's get @erwanor's blessing here

**Comment by TalDerei** [comment]
@jessepinho can you add a changeset?

File: apps/minifront/src/state/dutch-auction/get-sub-auctions.ts
Comment (jessepinho) [reply apps/minifront/src/state/dutch-auction/get-sub-auctions.ts]: Done. (Why this change?)

Comment (grod220) [reply apps/minifront/src/state/dutch-auction/get-sub-auctions.ts]: My local IDE was flagging it as breaking this rule: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-named-as-default-member.md. Think it's probably a good rule, not sure why `pnpm lint` doesn't catch it though 🤔

**Comment by jessepinho** [comment]
@TalDerei 

> can you add a changeset?

Good call, done

File: apps/minifront/src/state/dutch-auction/get-poisson-distribution.ts
Comment (jessepinho) [reply apps/minifront/src/state/dutch-auction/get-poisson-distribution.ts:7]: Hey @erwanor just a reminder that we need your eyes on this before it can merge

**Comment by TalDerei** [comment]
@jessepinho https://github.com/penumbra-zone/web/issues/1129 is especially relevant here!

**Comment by TalDerei** [comment]
> (We're going to need some help with the transaction approval UI, I think...)

@jessepinho https://github.com/penumbra-zone/web/issues/1129 is especially relevant here!

**Comment by TalDerei** [comment]
> * Filter out auction NFTs from the auction/swap asset selectors

References https://github.com/penumbra-zone/web/pull/1123

File: apps/minifront/src/state/dutch-auction/get-poisson-distribution.ts
Comment (TalDerei) [reply apps/minifront/src/state/dutch-auction/get-poisson-distribution.ts:7]: can you add a description here of the utility of a Poisson distribution in relation to the auction mechanism, eg. something like randomness is used to ensure the start heights are spaced out in a random yet statistically predictable manner

Comment (TalDerei) [reply apps/minifront/src/state/dutch-auction/get-poisson-distribution.ts:7]: the implementation looks correct to me.

**Review by TalDerei — APPROVED** [review]

**Review by erwanor — APPROVED** [review]
LGTM

**Pull request merged**
